state-and-event-handling/updating-state.md

```diff
@@ -0,0 +1,442 @@
+
+![A web app that lists three students. One student's name is in red. Two students' names are in green.](../assets/state-and-event-handling_updating-state_some-green-1.png)  
+_Fig. Sofia's ideal end result. Ada has a red name because they are absent. Soo-ah and Chrissy have green names because they are present._
+
+She'll need to update her app in the following ways:
+
+1. Create two CSS classes: one that sets the name to green, and another to set the name to red
+1. Create a piece of state in `Student` to hold if the student is present
+1. Create a button in `Student` to toggle if the student is present
+1. Create an event handler that updates state whenever the button is clicked
+1. Use the CSS class in `Student`. One CSS class should be applied if the student is present, and the other CSS class should be applied if they're absent.
+
+### CSS Classes
+
+In `src/components/Student.css`, Sofia adds these two classes:
+
+```css
+.green {
+  background-color: greenyellow;
+}
+
+.red {
+  background-color: lightcoral;
+}
+```
+
+### Create `isPresent` State in `Student`
+
+In `src/components/Student.js`, Sofia adds these two lines to create `isPresent` state:
+
+<!-- prettier-ignore-start -->
+```js
+import { useState } from 'react';
+```
+<!-- prettier-ignore-end -->
+
+This line is in the `Student` component function, before the return statement:
+
+<!-- prettier-ignore-start -->
+```js
+    const [isPresent, setIsPresent] = useState(false);
+```
+<!-- prettier-ignore-end -->
+
+### Create the Toggle Presence Button in `Student`
+
+Sofia updates the returned JSX in `Student`. She adds this `<button>`, with an appropriate label.
+
+<!-- Simon note: I added the indentation to clue to the student that it is likely indented? -->
+<!-- prettier-ignore-start -->
+```js
+    <button>Toggle if {props.name} is present</button>
+```
+<!-- prettier-ignore-end -->
+
+### Create the Event Handler
+
+Sofia now creates the event handler, which updates the `isPresent` state. This function is in the `Student` component function, before the return statement.
+
+<!-- Simon note: I added the indentation to clue to the student that it is likely indented? -->
+<!-- prettier-ignore-start -->
+```js
+    const togglePresence = () => {
+        setIsPresent(!isPresent);
+    }
+```
+<!-- prettier-ignore-end -->
+
+#### Attach the Event Handler to the Button
+
+She modifies her `<button>` JSX to include this click event listener:
+
+<!-- Simon note: I added the indentation to clue to the student that it is likely indented? -->
+<!-- prettier-ignore-start -->
+```js
+    <button onClick={togglePresence}>Toggle if {props.name} is present</button>
+```
+<!-- prettier-ignore-end -->
+
+### Modify the JSX to Use Conditional CSS Classes
+
+Now, Sofia wants to use conditional CSS classes.
+
+In her `Student` component, she creates a variable which holds the value of the appropriate CSS class. This value is based on:
```

Comment:
she wants to pick a css class, not specifically use "conditional classes"
this section sounds like the setup before she does it a few passages later, so future sounds better here.
```suggestion
Now, Sofia needs to conditionally select which CSS class to apply to the student's name.

In her `Student` component, she will create a variable to hold the value of the appropriate CSS class. This value will use:
```